operations/host_spawn.go

```diff
@@ -792,3 +796,375 @@ func hostRunCommand() cli.Command {
+			hostname = model.FromStringPtr(h.HostURL)
+			catcher.ErrorfWhen(hostname == "", "could not find hostname for host '%s'", hostID)
+			return user, hostname, catcher.Resolve()
+		}
+	}
+	return "", "", errors.Errorf("could not find host '%s' in user's spawn hosts", hostID)
+}
+
+// sanityCheckRsync performs some basic sanity checks for the common case in
+// which you want to mirror two directories. The trailing slash means to
+// overwrite all of the contents of the destination directory, so we check that
+// they really want to do this.
+func sanityCheckRsync(localPath, remotePath string, pull bool) (ok bool, err error) {
+	localPathIsDir := strings.HasSuffix(localPath, string(os.PathSeparator))
+	remotePathIsDir := strings.HasSuffix(remotePath, string(os.PathSeparator))
+
+	if localPathIsDir && !pull {
+		ok, err := sanityPrompt(fmt.Sprintf("The local directory '%s' will overwrite any existing contents in the remote directory '%s'", localPath, remotePath))
+		if err != nil {
+			return false, errors.Wrap(err, "error while asking user to sanity check rsync command")
+		}
+		if !ok {
+			return false, nil
+		}
+	}
+	if remotePathIsDir && pull {
+		ok, err := sanityPrompt(fmt.Sprintf("The remote directory '%s' will overwrite any existing contents in the local directory '%s'", remotePath, localPath))
+		if err != nil {
+			return false, errors.Wrap(err, "error while asking user to sanity check rsync command")
+		}
+		if !ok {
+			return false, nil
+		}
+	}
+	return true, nil
+}
+
+// sanityPrompt prints the message and asks the user if they would like to
+// continue or exit.
+func sanityPrompt(msg string) (ok bool, err error) {
+	fmt.Printf(msg + " Continue? (y/n) ")
+	r := bufio.NewReader(os.Stdin)
+	input, err := r.ReadString('\n')
+	if err != nil {
+		return false, errors.Wrap(err, "could not read input")
+	}
+	input = strings.TrimSpace(input)
+	return input == "y" || input == "yes", nil
+}
+
+type rsyncOpts struct {
+	local                string
+	remote               string
+	user                 string
+	host                 string
+	makeRemoteParentDirs bool
+	excludePatterns      []string
+	shouldDelete         bool
+	pull                 bool
+	dryRun               bool
+}
+
+func (opts *rsyncOpts) validate() error {
+	catcher := grip.NewBasicCatcher()
+	catcher.NewWhen(opts.local == "", "local path must be specified")
+	catcher.NewWhen(opts.remote == "", "remote path must be specified")
+	return catcher.Resolve()
+}
+
+// buildRsyncCommand takes the given options and constructs an rsync command.
+func buildRsyncCommand(ctx context.Context, opts rsyncOpts) (*exec.Cmd, error) {
+	if err := opts.validate(); err != nil {
+		return nil, errors.Wrap(err, "invalid rsync options")
+	}
+
+	rsync, err := exec.LookPath("rsync")
+	if err != nil {
+		return nil, errors.Wrap(err, "could not find rsync binary in the PATH")
+	}
+
+	args := []string{"-a", "--no-links", "--no-devices", "--no-specials", "-hh", "-I", "-z", "--progress", "-e", "ssh"}
```

Comment:
Is there ever a time when these logs would end up somewhere other than on a user's local machine, like Splunk? If so, we should avoid `--progress`. If not, then it's fine to keep. (Shipping `--progress` to a log aggregator ends up as a bit of a mess.)

Reply:
No, this is intended only for users' convenience when copy/pasting files between their local machine and spawn host. There's no actual interaction with evergreen aside from getting the host info from the host ID.